client/blocks/product-selector/index.jsx

```diff
@@ -193,8 +197,24 @@ export class ProductSelector extends Component {
 	};
 
 	handleProductOptionSelect( stateKey, productSlug ) {
+		const { intervalType } = this.props;
+		const relatedStateChange = {};
+		const otherInterval = 'yearly' === intervalType ? 'monthly' : 'yearly';
+		const productId = this.getProductIdFromStateKey( stateKey );
```

Comment:
What if we passed a `product` as a third argument to `handleProductOptionSelect`? This way we could access `product.id` directly and get rid of the `getProductIdFromStateKey` method along with the regular expression.

To prevent a potential confusion, the `productSlug` could be renamed to something clearer like `selectedProductSlug`. 

The reason is that I feel that the regular expression may be a bit fragile and prone to any inconsistencies in the product's naming convention. I know it's not the case right now, so feel free to ignore this suggestion altogether.

Reply:
I actually like that suggestion, nice 👍 Will simplify things a little.

FWIW, it will be hard to come up with a bug coming from the product naming convention - what we're doing here relies solely on how we define the state keys, and that's uniquely defined in this component. However, I agree it is a bit tricky and convoluted, so happy to simplify.

So, I'll just go with passing `product.id` directly, as we won't be needing the entire `product` object.

Reply:
Thanks @delawski, addressed in https://github.com/Automattic/wp-calypso/pull/37485/commits/86681757228dd2b9d59bb1f8851be9a34c951373.